File: bookkepping-web/apps/web-antd/src/api/core/budget.ts
import { requestClient } from '#/api/request';

export namespace BudgetApi {
  /** 预算详情结果 */
  export interface BudgetResult {
    id: number;
    userId: number;
    username: string;
    amount: number;
    budgetMonth: string;
    startDate: string;
    endDate: string;
    createdAt: string;
    updatedAt: string;
    spendingTotalAmount: number;
  }

  /** 更新预算参数 */
  export interface UpdateBudgetParams {
    amount: number;
  }
}

/**
 * 获取预算详情
 */
export async function getBudgetDetailApi() {
  return requestClient.get<BudgetApi.BudgetResult>('/budget');
}

/**
 * 更新预算
 */
export async function updateBudgetApi(params: BudgetApi.UpdateBudgetParams) {
  return requestClient.put('/budget', params);
}

File: api-server/src/main/java/com/bookkeeping/entity/model/Budget.java
package com.bookkeeping.entity.model;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;

/**
 * <p>
 * 预算表
 * </p>
 *
 * @author bookkeeping
 * @since 2025-04-19
 */
@Getter
@Setter
@Schema(name = "Budget", description = "预算表")
public class Budget implements Serializable {

    private static final long serialVersionUID = 1L;

    @Schema(description = "主键ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @Schema(description = "用户 user.id")
    private Long userId;

    @Schema(description = "用户名")
    private String username;

    @Schema(description = "预算金额（分）")
    private Long amount;

    @Schema(description = "预算月份 yyyy-MM格式")
    private String budgetMonth;

    @Schema(description = "预算开始日期")
    private LocalDate startDate;

    @Schema(description = "预算结束日期")
    private LocalDate endDate;

    @Schema(description = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @Schema(description = "修改时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
